### ISSUE TITLE
Failed to construct 'URL' when schema is provided for POST path

### ISSUE THREAD

**Issue by piotrwalczak1** [swagger-api/swagger-ui#9215]
- OS: Linux
 - Browser: chrome
 - Version: 4.14.0
 - Method of installation: npm
 - Swagger-UI version: 4.14.0
 - Swagger/OpenAPI version: 3.0

### Content & configuration

Example Swagger/OpenAPI definition:
```yaml
openapi: "3.0.0"
info:
  title: "whatever"
  version: "1.0.0"
servers:
  - url: "https://sample-url.com/dev"
paths:
  /deprovisioning:
    post:
      requestBody:
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/deprovisioningModelBodyValidator"
        required: true
  /provisioning:
    post:
      requestBody:
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/provisioningModelBodyValidator"
        required: true

components:
  schemas:
    deprovisioningModelBodyValidator:
      type: object
      properties:
        thingName:
          type: string
      required:
        - thingName
    provisioningModelBodyValidator:
      type: object
      properties:
        macAddress:
          type: string
      required:
        - macAddress

```

Swagger-UI configuration options:
```js
import SwaggerUI from "swagger-ui";
import "swagger-ui/dist/swagger-ui.css";

// const spec = require('./openapi.yaml');
import * as spec from "./openapi.yaml";

const ui = SwaggerUI({
  spec,
  dom_id: "#swagger",
});

ui.initOAuth({
  appName: "Swagger UI Webpack Demo",
  // See https://demo.identityserver.io/ for configuration details.
  clientId: "implicit",
});

```

### Describe the bug you're encountering
I am getting the Failed to construct 'URL': Invalid URL when trying to use the webpack example with the yaml provided. As soon as I click the endpoint path, I am getting the error. The config I provided is perfectly fine when using online swagger editor.

### Expected behavior
It should work, as yaml is in open api standard

### Screenshots
![image](https://github.com/swagger-api/swagger-ui/assets/95431451/976e40e3-92be-48ec-b2ad-dbf1d68b9848)
![image](https://github.com/swagger-api/swagger-ui/assets/95431451/0f599ff1-fe0d-43f9-bcc9-5ef6f7fcaa8b)

**Comment by GonzaOrtiz** [comment]
Could anyone solve it? I have the same error

**Comment by char0n** [comment]
You either have to pin swagger-client to https://github.com/swagger-api/swagger-js/releases/tag/v3.20.2 or use SwaggerUI@5.

**Comment by steph-lion** [comment]
Same problem with `swagger-ui-express`

**Comment by char0n** [comment]
All rihgt, seems like a serious regression. I’ll introduce backward compatible patch to swagger-client to allow EOL swageer-ui versions to function properly without pinning swagger-client version.

**Comment by bikashburma615** [comment]
same problem with flask_swagger_ui

**Comment by char0n** [comment]
This issue have been addressed in [swagger-client@3.22.2](https://github.com/swagger-api/swagger-js/releases/tag/v3.22.2). If you reinstall your SwaggerUI version, the latest swagger-client will be installed as transitive dependency and will resolve the issue.

**Comment by steph-lion** [comment]
> This issue have been addressed in [swagger-client@3.22.2](https://github.com/swagger-api/swagger-js/releases/tag/v3.22.2). If you reinstall your SwaggerUI version, the latest swagger-client will be installed as transitive dependency and will resolve the issue.

Should I wait for [this](https://www.npmjs.com/package/swagger-ui-express) npm module to release its new version for this to be resolved?

**Comment by char0n** [comment]
> > This issue have been addressed in [swagger-client@3.22.2](https://github.com/swagger-api/swagger-js/releases/tag/v3.22.2). If you reinstall your SwaggerUI version, the latest swagger-client will be installed as transitive dependency and will resolve the issue.
> 
> Should I wait for [this](https://www.npmjs.com/package/swagger-ui-express) npm module to release its new version for this to be resolved?

Nope, no need to release anything here. Just re-installing this npm module (swagger-ui/swagger-ui-react)  will automatically install latest swagger-client .
